DEVELOPMENT.md

```diff
@@ -38,4 +38,20 @@ for gapic-generator-java's Bazel build.
 
     ```sh
     mvn fmt:format
-    ```
\ No newline at end of file
+    ```
+
+## Testing guide
```

Comment:
This section will be really helpful for new contributors to the repo! 🎉 

One quick thought from working on https://github.com/googleapis/sdk-platform-java/pull/1720 earlier that required updating a bunch of golden files - could it be helpful to link to the different locations/further instructions pertaining to running and updating these layers as well? A couple off the top of my head:

Unit test goldens ([gapic-generator-java/DEVELOPMENT.md section](https://github.com/googleapis/sdk-platform-java/blob/main/gapic-generator-java/DEVELOPMENT.md#unit-tests))
`mvn test -DupdateUnitGoldens`

Integration test goldens ([gapic-generator-java/DEVELOPMENT.md section ](https://github.com/googleapis/sdk-platform-java/blob/main/gapic-generator-java/DEVELOPMENT.md#integration-tests))
`bazelisk run //test/integration:update_redis `(repeat for all modules [here](https://github.com/googleapis/sdk-platform-java/tree/main/test/integration/goldens))

Showcase test goldens ([showcase/README.md](https://github.com/googleapis/sdk-platform-java/tree/main/test/integration/goldens#how-to-update-goldens))
`mvn compile -P update`

Reply:
Updated, PTAL.

Reply:
LGTM - thanks!